rat/compiler.py

```diff
@@ -133,362 +384,190 @@ def _identify_primary_symbols(self):
+                                )
 
-        for top_expr in self.expr_tree_list:
-            primary_symbol = self._get_primary_symbol_from_statement(top_expr)
-            # Identify the primary dataframe if there is one and resolve naming issues
-            primary_key = primary_symbol.get_key()
-
-            match primary_symbol:
-                case ops.Data():
-                    # If the primary primary_symbol is data, the reference dataframe is the data dataframe
-                    primary_df = self.data_df
-                # TODO: This isn't implemented but might make sense
-                # case ops.Subscript():
-                # If the primary primary_symbol is a subscript, the reference dataframe is the data dataframe
-                #    primary_df = self.data_df
-                case ops.Param():
-                    # For parameters
-                    # 1. A subscripted primary parameter primary_symbol must have a dataframe
-                    # 2. If there is no subscript, then:
-                    #    A. If a dataframe exists, then column names must be uniquely defined
-                    #    B. If no dataframe exists, the parameter is a scalar
-                    # Since the lines are evaluated in a topological order, the rows of the primary dataframe are known at this point.
-                    if primary_key in self.parameter_base_df:
-                        primary_df = self.parameter_base_df[primary_key].copy()
-
-                        # If the primary primary_symbol is subscripted, should be a dataframe
-                        if primary_symbol.subscript:
-                            # Handle subscript renaming
-                            try:
-                                primary_df.columns = tuple(primary_symbol.subscript.get_key())
-                            except ValueError as e:
-                                # if usage subscript length and declared subscript length do not match, raise exception
-                                msg = f"Rename {primary_symbol.code()} not possible, {len(primary_df.columns)} required names but {len(primary_symbol.subscript.get_key())} given."
-                                raise CompileError(
-                                    msg, self.model_code_string, primary_symbol.line_index, primary_symbol.column_index
-                                ) from e
-
-                            self.parameter_base_df[primary_key].columns = primary_df.columns
+                            else:
+                                for index, subscript_column in enumerate(primary_variable_record.subscript_alias):
+                                    if subscript_column not in self.data_df_columns:
+                                        # If there's a subscript name that's not present in the input dataframe, this means
+                                        # it's a subscript "alias", which points to the "true" subscripts that's already on
```

Comment:
Yes, it's a rename. "aliases" are the column names of the base dataframe of that parameter. Looking back, it does sound confusing, but I thought them as aliases since your're renaming a set of input dataframe columns in to a single name.